.github/workflows/main.yml

```
@@ -64,10 +64,40 @@ jobs:
 
-      - name: Run pytest
+      - name: Test docs build
         run: |
           pixi run -e ${{ matrix.env }} doc-build
 
+  ################################################################################################
+  # Publish-Docs: Publish documentation to GitHub Pages using mike only on main branch
+  ################################################################################################
+  Publish-Docs:
+    needs: Build-Docs
+    if: github.ref == 'refs/heads/main' 
```

yamllint (1.35.1)

[error] trailing spaces

(trailing-spaces)